components/org.wso2.carbon.identity.authz.spicedb/src/main/java/org/wso2/carbon/identity/authz/spicedb/handler/spicedb/SpiceDbPermissionRequestsHandler.java

```diff
@@ -0,0 +1,85 @@
+import org.json.JSONObject;
+import org.wso2.carbon.identity.authz.spicedb.constants.SpiceDbConstants;
+import org.wso2.carbon.identity.authz.spicedb.handler.model.ErrorResponse;
+import org.wso2.carbon.identity.authz.spicedb.handler.model.PermissionBulkCheckResponse;
+import org.wso2.carbon.identity.authz.spicedb.handler.model.PermissionCheckResponse;
+import org.wso2.carbon.identity.authz.spicedb.handler.util.SpiceDbHttpHandler;
+import org.wso2.carbon.identity.authz.spicedb.handler.util.SpiceDbRequestInterface;
+import org.wso2.carbon.identity.authz.spicedb.handler.util.SpiceDbResponseInterface;
+
+/**
+ * This class contains methods that handle  all authorization related requests from SpiceDB.
+ */
+public class SpiceDbPermissionRequestsHandler {
+    private static final SpiceDbHttpHandler SPICEDB_HTTP_HANDLER = new SpiceDbHttpHandler();
+
+    public SpiceDbResponseInterface checkAuthorization(SpiceDbRequestInterface authorizationRequest) {
+
+        JSONObject object;
+        try {
+            HttpResponse response = SPICEDB_HTTP_HANDLER.sendPOSTRequest(SpiceDbConstants.BASE_URL +
+                            SpiceDbConstants.PERMISSION_CHECK, authorizationRequest.parseToJSON());
+            object = new JSONObject(SPICEDB_HTTP_HANDLER.parseToString(response));
+            int statusCodeType = response.getStatusLine().getStatusCode() / 100;
+            switch (statusCodeType) {
+                case 2:
+                case 3:
+                    return new PermissionCheckResponse(object);
+                case 4:
+                case 5:
+                    return new ErrorResponse(object);
+                default:
+                    return new ErrorResponse(new JSONObject("{\n\"message\": \"Cannot identify error code.\"\n\"" +
+                            "details\": \"Cannot provide details. Please check spiceDB connection.\""));
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("An error occurred while trying to check permission from spiceDB.", e);
+        }
+    }
+
+    public SpiceDbResponseInterface bulkCheckAuthorization(SpiceDbRequestInterface bulkAuthorizationRequest) {
+        JSONObject object;
+        try {
+            HttpResponse response = SPICEDB_HTTP_HANDLER.sendPOSTRequest(SpiceDbConstants.BASE_URL +
+                    SpiceDbConstants.PERMISSIONS_BULKCHECK, bulkAuthorizationRequest.parseToJSON());
+            object = new JSONObject(SPICEDB_HTTP_HANDLER.parseToString(response));
+            int statusCodeType = response.getStatusLine().getStatusCode() / 100;
+            switch (statusCodeType) {
+                case 2:
+                case 3:
+                    return new PermissionBulkCheckResponse(object);
+                case 4:
+                case 5:
+                    return new ErrorResponse(object);
+                default:
+                    return new ErrorResponse(new JSONObject("{\n\"message\": \"Cannot identify error code.\"\n\"" +
```

Comment:
create json object and pare

Reply:
Resolved here:
 [Change error handling and improve code.](https://github.com/wso2-extensions/identity-authz-spicedb/pull/1/commits/ace19335123cd3ff44e6776136c046f39934f220)